lib/token_store/sql_store.rb

```diff
@@ -28,12 +33,29 @@ def delete(token)
       record.destroy!
     end
 
+    def delete_all_for_user(userid)
+      user = find_user_by_userid(userid)
+      user.sessions.where(Session.arel_table[:session_id].matches("#{@namespace}%")).destroy_all
```

Comment:
> could you use the 3 parameter matches here?

I don't understand the question.  I think I get what you are suggesting via a commit suggestion, but I am not sure what this question is asking.

> is there a way to not use a match at all?

We generate tokens for the API that look like this:

```ruby
"MIQ:TOKENS:API:#{token}"
"MIQ:TOKENS:UI:#{token}"
"MIQ:TOKENS:WS:#{token}"
```

So, if I want to remove a for a given namespace, this is required.  This is why I am doing a `TokenStore.token_caches.each` because each of the caches are configured via the namespace (`"API"`, `"UI"`, and `"WS"`) and blowing them all away without the matches would be going outside of the scope of each of those `TokenStore` instances.

Reply:
> > could you use the 3 parameter matches here?
> 
> I don't understand the question. ...

Okay, looking again I think I get it now, and I assume you mean passing `, nil, true` to `.matches` like you had shown above.

Can provide a reason as to why you prefer that?

Reply:
since you are using `downcase` the comparison is case sensitive - so the `nil, true` uses `LIKE` over `ILIKE`
we are probably not using indexes anyway, so it may be moot.

Reply:
I can change it, it is just nice having some kind of record of "why".

This should be a low traffic endpoint, so not expecting performance to be an issue, but can do it anyway.